index.js

```diff
@@ -72,22 +71,25 @@ function findAllDependencies(file, knownDependencies, sourceDirectories, knownFi
 }
 
 // Given a source directory (containing top-level Elm modules), locate the
-// elm-package.json file that includes it.
-function findElmPackage(baseDir, currentDir) {
+// elm-package.json file that includes it and get all its source directories.
+function getElmPackageSourceDirectories(baseDir, currentDir) {
   if (typeof currentDir === "undefined") {
     currentDir = baseDir;
   }
 
   var elmPackagePath = path.join(currentDir, 'elm-package.json');
   if (fs.existsSync(elmPackagePath)) {
-    return elmPackagePath;
+    var sourceDirectories = getSourceDirectories(elmPackagePath);
+    if (_.includes(sourceDirectories, baseDir)) {
```

Comment:
I'm worried that this never halts, when this isn't true. I think this should raise an error. I also think that the source-directories don't have to include base dir right? You could have a elm-package with only `./src`.